android/ui/src/main/java/com/trafi/ui/RouteSearchTab.kt

```diff
@@ -0,0 +1,151 @@
+package com.trafi.ui
+
+import androidx.compose.foundation.*
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Card
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.tooling.preview.Preview
+import com.trafi.ui.component.internal.RouteSearchTabConstants
+import com.trafi.ui.theme.MaasTheme
+import com.trafi.ui.theme.currentTheme
+import com.trafi.ui.theme.darkColors
+
+@Composable
+private val constants
+    get() = RouteSearchTabConstants(currentTheme)
+
+data class TabItem(
+    val id: String,
+    val icon: String,
+    val active: Boolean,
+    val duration: String?,
+    val price: String?,
+)
+
+@Composable
+fun RouteSearchTab(
+    tabItem: TabItem,
+    isSelected: Boolean,
+    modifier: Modifier = Modifier,
+    onRouteTabClick: () -> Unit,
+) {
+    Card(
+        shape = RoundedCornerShape(constants.borderRadius),
+        border = BorderStroke(constants.borderWidth,
+            color = if (isSelected) constants.borderColorActive else constants.defaultBackgroundColor),
+        modifier = modifier
+            .padding(constants.padding)
+            .widthIn(min = constants.minTabWidth)
+            .heightIn(min = constants.minTabHeight)
+    ) {
+        Box(
+            modifier = modifier
```

Comment:
The incoming `modifier` is applied twice - for the `Card` and again for the `Box`

Reply:
@justasm This decision was made not accidentally because, when I tested with modifiers it needed same parameters for both layouts, but now I realised that it's not good, because with some attributes it will broke the layout, so I will fix this